### PR TITLE
HP-2047 New HDS into use

### PR DESCRIPTION
**Pull request by Riippi**
- Update HDS to newest version and change components where it's necessary
- Update translations and colors
- Update some other dependencies
- Add .nvmrc file

https://helsinkisolutionoffice.atlassian.net/browse/HP-2047

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
 
       - uses: actions/setup-node@v2
         with:
-          node-version: '14.19.1'
```
Comment (NikoHelle) [inline .github/workflows/CI.yml]: There was discussion about the version in 

https://helsinkicity.slack.com/archives/C01RNQ5PV5Z/p1702026731704299

Might be best to upgrade to node 18 or even 20.

Ps. this should match the version in .nvmrc?

File: .vscode/settings.json [hunk 2]
```diff
@@ -5,5 +5,8 @@
     { "language": "typescript", "autoFix": true },
     { "language": "typescriptreact", "autoFix": true }
   ],
-  "typescript.tsdk": "node_modules/typescript/lib"
-}
\ No newline at end of file
+  "typescript.tsdk": "node_modules/typescript/lib",
```
Comment (NikoHelle) [inline .vscode/settings.json]: This file was also added to .gitignore so this can be removed.

**Review by NikoHelle — CHANGES_REQUESTED** [review]
Nice upgrade!

Node version is also in Dockerfile, so that must also be upgraded to match other node versions.

The base-file used there, helsinkitest/node, should not be used, but ubi-images mentioned in https://helsinkicity.slack.com/archives/C01RNQ5PV5Z/p1702026731704299

The Dockerfile also uses ancient nginx 1.2. which could be upgraded too
https://helsinkicity.slack.com/archives/C01RNQ5PV5Z/p1707122854109759

File: .vscode/settings.json (hunk 2 above)
Comment (Riippi) [reply .vscode/settings.json]: Removed the file.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (Riippi) [reply .github/workflows/CI.yml]: It's now 16.20.2 until the "dependency can of worms" is handled in other task.

**Comment by NikoHelle** [comment]
In my Chrome the app has weird borders and always visible scrollbar

Local:
image

In dev:
image

There is some css missing, as the body has default 8px margin. That is overridden in dev-version.

Could be the problem that Tuomo has been fixing lately.

**Comment by NikoHelle** [comment]
Clicking the internal link "Evästeasetukset" reloads the whole app again. That is prevented in the dev version (old footer)

**Comment by NikoHelle** [comment]
Clicking the menu in mobile view does nothing.

**Comment by NikoHelle** [comment]
Is there is a reason why English is the first primary language? Finnish is most likely the most used language.

File: Dockerfile [hunk 3]
```diff
@@ -58,7 +58,7 @@ COPY . /app
 RUN yarn build
 
 # =============================
-FROM registry.access.redhat.com/ubi8/nginx-120 as production
+FROM registry.access.redhat.com/ubi9/nginx-122 AS production
```
Comment (NikoHelle) [inline Dockerfile]: 1.22. is said to be too old in this discussion:
https://helsinkicity.slack.com/archives/C01RNQ5PV5Z/p1707122854109759. 

Can we used newer/latest?

**Review by NikoHelle — CHANGES_REQUESTED** [review]
Few UI bugs should be fixed...

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr306-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr306-ui.dev.hel.ninja** :rocket::rocket::rocket:

File: Dockerfile (hunk 3 above)
Comment (Riippi) [reply Dockerfile]: 1.22 is the newest that redhat offers and should be ok. This was discussed a bit here: https://helsinkicity.slack.com/archives/C01RNQ5PV5Z/p1707311323479389?thread_ts=1707310835.048119&cid=C01RNQ5PV5Z

**Comment by Riippi** [comment]
> Few UI bugs should be fixed...

- 8px padding was becaue hds removed normalise. Now this is fixed in css
- "Evästeasetukset" link is now router link and doesn't reload whole page. Also "saatavuusseloste" fixed.
- mobile menu button is fixed
- Finnish is now default language and automatic browser check is removed for now

**Comment by Riippi** [comment]
Added test for header language change so the test coverage is over 65%

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr306-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr306-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr306-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by NikoHelle** [comment]
Good fixes! Couple of issues

The ingress text should not be bold according to designs:

image

Finnish should be the first language option
image

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr306-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Review by NikoHelle — APPROVED** [review]
All done!

Something I forget maybe 50% time when updating: set new version to package.json! 

It will show in https://profiili.dev.hel.ninja/env-config.js and then you can check what version each env is running.

You can freely update it to any version number you want, nothing is depending on it. :)

**Pull request merged**